recipes/ffmpeg/all/conanfile.py

```diff
@@ -0,0 +1,352 @@
+        args.append("--enable-libmp3lame" if self.options.mp3lame else "--disable-libmp3lame")
+        args.append("--enable-libfdk-aac" if self.options.fdk_aac else "--disable-libfdk-aac")
+        args.append("--enable-libwebp" if self.options.webp else "--disable-libwebp")
+        args.append("--enable-openssl" if self.options.ssl == "openssl" else "--disable-openssl")
+
+        if self.options.x264 or self.options.x265 or self.options.postproc:
+            args.append("--enable-gpl")
+
+        if self.options.fdk_aac:
+            args.append("--enable-nonfree")
+
+        if self.settings.os == "Linux":
+            args.append("--enable-alsa" if self.options.alsa else "--disable-alsa")
+            args.append("--enable-libpulse" if self.options.pulse else "--disable-libpulse")
+            args.append("--enable-vaapi" if self.options.vaapi else "--disable-vaapi")
+            args.append("--enable-vdpau" if self.options.vdpau else "--disable-vdpau")
+            if self.options.xcb:
+                args.extend(["--enable-libxcb", "--enable-libxcb-shm",
+                             "--enable-libxcb-shape", "--enable-libxcb-xfixes"])
+            else:
+                args.extend(["--disable-libxcb", "--disable-libxcb-shm",
+                             "--disable-libxcb-shape", "--disable-libxcb-xfixes"])
+
+        if self.settings.os == "Macos":
+            args.append("--enable-appkit" if self.options.appkit else "--disable-appkit")
+            args.append("--enable-avfoundation" if self.options.avfoundation else "--disable-avfoundation")
+            args.append("--enable-coreimage" if self.options.avfoundation else "--disable-coreimage")
+            args.append("--enable-audiotoolbox" if self.options.audiotoolbox else "--disable-audiotoolbox")
+            args.append("--enable-videotoolbox" if self.options.videotoolbox else "--disable-videotoolbox")
+            args.append("--enable-securetransport" if self.options.ssl == "securetransport" else "--disable-securetransport")
+
+        if self.settings.os == "Windows":
+            args.append("--enable-libmfx" if self.options.qsv else "--disable-libmfx")
+
+        # FIXME disable CUDA and CUVID by default, revisit later
+        args.extend(["--disable-cuda", "--disable-cuvid"])
+
+        # ffmpeg"s configure is not actually from autotools, so it doesn"t understand standard options like
+        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        self._patch_sources()
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
```

Comment:
Please model the `.pc` files in `package_info`...

Reply:
Shall I model using component here?